### DISCUSSION TITLE
posts/k3s-ha-install/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#193] [Comments]
# posts/k3s-ha-install/

Documentation for Techno Tim YouTube videos for all your copy pasta needs!

https://technotim.live/posts/k3s-ha-install/

**Comment by rayk32** [comment]
I've tried these instructions a few times and I can't get them to work.

The following isn't clear:

---------
export K3S_DATASTORE_ENDPOINT='mysql://username:password@tcp(database_ip_or_hostname:port)/database'

I assume username, password, IP address and port need to be changed to match my environment.  Should /database be changed to the actual database name?

---------
It is unclear which command needs to be run on the other master servers.  Both commands fail to start the service.

curl -sfL https://get.k3s.io | sh -s - server --node-taint CriticalAddonsOnly=true:NoExecute --tls-san load_balancer_ip_or_hostname

or

curl -sfL https://get.k3s.io | sh -s - server --token=SECRET --node-taint CriticalAddonsOnly=true:NoExecute --tls-san load_balancer_ip_or_hostname

Both commands fail with the following message:

[INFO]  systemd: Starting k3s
Job for k3s.service failed because the control process exited with error code.
See "systemctl status k3s.service" and "journalctl -xe" for details.
inet@k3s-master-2:~$ systemctl status k3s.service
● k3s.service - Lightweight Kubernetes
     Loaded: loaded (/etc/systemd/system/k3s.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Tue 2022-11-15 07:02:02 UTC; 5s ago
       Docs: https://k3s.io
    Process: 22396 ExecStartPre=/bin/sh -xc ! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service (code=exited, status=0/SUCCESS)
    Process: 22406 ExecStartPre=/sbin/modprobe br_netfilter (code=exited, status=0/SUCCESS)
    Process: 22407 ExecStartPre=/sbin/modprobe overlay (code=exited, status=0/SUCCESS)
    Process: 22408 ExecStart=/usr/local/bin/k3s server --token=K10447c64a7096f8d169f65a873dfb8d7fe233d94806d9afffedff7b66447dd16d6::server:eadb5a9f094a8456c35bf1208228384f --node-taint Critic>
   Main PID: 22408 (code=exited, status=1/FAILURE)

Nov 15 07:02:02 k3s-master-2 systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
Nov 15 07:02:02 k3s-master-2 systemd[1]: k3s.service: Failed with result 'exit-code'.
Nov 15 07:02:02 k3s-master-2 systemd[1]: Failed to start Lightweight Kubernetes.
Nov 15 07:02:07 k3s-master-2 systemd[1]: k3s.service: Scheduled restart job, restart counter is at 3.
Nov 15 07:02:07 k3s-master-2 systemd[1]: Stopped Lightweight Kubernetes.
Nov 15 07:02:07 k3s-master-2 systemd[1]: Starting Lightweight Kubernetes...

Please advise.

**Comment by jsparham** [comment]
While setting up nginx, you may come across an error:
/etc$ sudo rc-service nginx start
 * Caching service dependencies ...                                                                                                                                    [ ok ]
nginx: [emerg] dlopen() "/usr/lib/nginx/modules/ngx_stream_module.so" failed (Error loading shared library /usr/lib/nginx/modules/ngx_stream_module.so: No such file or directory) in /etc/nginx/nginx.conf:2
nginx: configuration file /etc/nginx/nginx.conf test failed
 * ERROR: nginx failed to start

install nginx-mod-stream to correct it. Using alpine Linux, it was sudo apk add nginx-mod-stream.
